tests/golang.rs

```diff
@@ -24,7 +25,92 @@ macro_rules! tests {
 
         // https://github.com/golang/go/blob/master/test/chan/doubleselect.go
         mod doubleselect {
-            // TODO
+            use super::*;
+
+            const ITERATIONS: usize = 100_000;
+
+            #[test]
+            #[ignore]
+            fn test_double_select() {
+                let (c1, r1) = channel::unbounded();
+                let (c2, r2) = channel::unbounded();
+                let (c3, r3) = channel::unbounded();
+                let (c4, r4) = channel::unbounded();
+                let (done_s, done_r) = channel::unbounded();
+                let (mux_s, mux_r) = channel::unbounded();
+
+                fn mux(out: channel::Sender<usize>, in_c: channel::Receiver<usize>, done: channel::Sender<bool>) {
+                    for val in in_c.0 {
+                        out.send(val);
+                    }
+
+                    done.send(true);
```

Comment:
Let's just do `drop(out)` right before this line so that receiving from `done` implies `out` was dropped.

Reply:
Ok, will update.

Reply:
Looks like `drop(out)` was added below `done.send(true)`, while it should be above. :)